tests/test_report_crud.py

```diff
@@ -0,0 +1,115 @@
+import pry
+import json
+from tests.BaseCase import BaseCase
+
+class TestReportCrud(BaseCase):
+    # GET single
+    def test_successful_get_record(self):
+        # Given
+        payload = {
+        "salary": 55000.0,
+        "zipcode": 11111,
+        "credit_score": 695,
+        "monthly_debt": 1100.0,
+        "downpayment_savings": 10000.0,
+        "downpayment_percentage": 10.0,
+        "rent": 1800.0,
+        "goal_principal": 0.0
+        }
+
+        response1 = self.app.post('/api/v1/report', headers={"Content-Type": "application/json"}, data=json.dumps(payload))
+        id = response1.json['data']['id']
+        url = f'/api/v1/report/{id}'
+
+        # When
+        get_response = self.app.get(url, headers={"Content-Type": "application/json"})
+        # Then
+        self.assertEqual(200, get_response.status_code)
+        # self.assertEqual(id, )
+
+    # CREATE
+    def test_successful_post_record(self):
+        # Given
+        payload = {
+          "salary": 55000.0,
+          "zipcode": 11111,
+          "credit_score": 695,
+          "monthly_debt": 1100.0,
+          "downpayment_savings": 10000.0,
+          "downpayment_percentage": 10.0,
+          "rent": 1800.0,
+          "goal_principal": 0.0
+        }
+        # When
+        response1 = self.app.post('/api/v1/report', headers={"Content-Type": "application/json"}, data=json.dumps(payload))
+        id = response1.json['data']['id']
+        url = f'/api/v1/report/{id}'
+
+        response2 = self.app.get(url, headers={'Content-Type': "application/json"})
+
+        body = response2.json
+
+        # Then
+        self.assertEqual(200, response2.status_code)
+        self.assertEqual(str, type(id))
```

Comment:
We can test the confirmation hash information.
I believe ln 54 may be testing the wrong id or section. I think it should be `self.assertEqual(str, body['data']['id'])` because `id` is actually coming from response1.